### PR TITLE
Improve state handling in cases of memory allocation failure

### PR DESCRIPTION
**Pull request by AtariDreams**
Many lines of code assume that malloc will not fail. In cases where it does, sometimes the program does not know, and as a result, memory can leak and more disastrous consequences can happen before the program ultimately finds something is wrong and then calls exit();

This PR aims to better handle memory in these cases

### CODE REVIEW

**Comment by AtariDreams** [comment]
Also, did you know what checking for NULL before realloc is not necessary? If ptr is NULL, realloc will behave like malloc.

**Comment by michaelrsweet** [comment]
@AtariDreams The `NULL` behavior with `realloc` wasn't part of C89, which is why so much of the CUPS code protects against it. Some semi-supported platforms (Solaris in particular) don't support this, so we'll be very conservative when accepting changes like this for CUPS 2.x. For CUPS 3.x we are targeting C99 and higher which cleans up a lot of legacy behaviors...

**Comment by AtariDreams** [comment]
Addressed @michaelrsweet

File: filter/rastertopwg.c [hunk 1]
```diff
@@ -152,6 +152,23 @@ main(int  argc,				/* I - Number of command-line args */
       return (1);
     }
 
+    if (inheader.cupsColorOrder != CUPS_ORDER_CHUNKED)
```
Comment (michaelrsweet) [inline filter/rastertopwg.c:155]: This seems unrelated to the memory allocation changes.

File: scheduler/cups-driverd.cxx [hunk 2]
```diff
@@ -152,7 +152,7 @@ static ppd_info_t	*add_ppd(const char *filename, const char *name,
 				 size_t size, int model_number, int type,
 				 const char *scheme);
 static int		cat_drv(const char *name, int request_id);
-static void		cat_ppd(const char *name, int request_id);
+static void		cat_ppd(const char *name, int request_id) _CUPS_NORETURN;
```
Comment (michaelrsweet) [inline scheduler/cups-driverd.cxx:155]: Unrelated.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
This PR seems to have a mix of changes in it and needs to be cleaned up.

**Pull request closed without merging**